### PR TITLE
Fix conversion from heterogeneous to homogeneous connector when deleting last synapse of a type

### PR DESCRIPTION
**Pull request by jakobj**
See #565 for a discussion of the issue. This PR fixes that issue and provides a test.
I suggest @sdiazpier and @suku248 as reviewers.

### CODE REVIEW

File: testsuite/unittests/issue-565.sli [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+
+ /* BeginDocumentation
+Name: testsuite::issue-565
+
+Synopsis: (issue-565) run -> NEST exits if test fails
+
+Description:
+This test provokes a switch from a heterogeneous connector to a homogeneous connector using the structural plasticity framework and asserts that the resulting homogeneous connector is correctly used.
```
Comment (janhahne) [inline testsuite/unittests/issue-565.sli]: This line could use a line break

File: testsuite/unittests/issue-565.sli [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+[2] [1] << /rule /all_to_all >> << /model /static_synapse /weight 0. >> Connect
+
+% Create second connection with new synapse type, causing the
+% homogeneous connector to be turned into a heterogeneous connector
+/static_synapse /dev_syn CopyModel
+[2] [3] << /rule /all_to_all >> << /model /dev_syn >> Connect
+
+% Delete the first connection again, causing the heterogeneous
+% connector to be turned into a homogeneous connector
+[2] cvgidcollection [1] cvgidcollection << /rule /all_to_all >> << /model /static_synapse >> Disconnect_g_g_D_D
```
Comment (janhahne) [inline testsuite/unittests/issue-565.sli]: Same here, maybe add a line break after `/all_to_all >>`

File: testsuite/unittests/issue-565.sli [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+100 Simulate
+
+% The spike detector should have recorded three events
+{
+  3 GetStatus /n_events get 3 eq
+} assert_or_die
```
Comment (janhahne) [inline testsuite/unittests/issue-565.sli]: I see from your example in the issue #565 how this ensures that everything works properly, but wouldn't it be much more fail-safe to add another simulation to this unit test where one neuron with the same parameters is just connected to the spike detector (without disconnecting a temporary connection) and then (after both simulations) to compare the actual spike times recorded by the spike detector?

File: testsuite/unittests/issue-565.sli [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+SeeAlso:
+*/
+
+(unittest) run
+/unittest using
+
+M_ERROR setverbosity
+
+% Create neurons and spike detector
+/iaf_neuron 2 << /I_e 400. >> Create /neurons Set
```
Comment (janhahne) [inline testsuite/unittests/issue-565.sli]: Use `iaf_psc_alpha` instead (see PR #526)

**Comment by jakobj** [comment]
Thanks @janhahne for the quick review. I have pushed a new version addressing all your comments.

**Comment by sdiazpier** [comment]
Dear @jakobj thanks for fixing this issue. I agree with the solution :+1:  from my side.

**Comment by heplesser** [comment]
@janhahne Could you confirm that @jakobj has addressed your concerns successfully?

**Comment by janhahne** [comment]
@heplesser @jakobj Yes, he did! Nice work :+1:

**Pull request merged**